### ISSUE TITLE
Inquiry about Closed Source Dependency org.mitre.caasd:commons-closed:1.0.1

### ISSUE THREAD

**Issue by dragonlayout** [mitre-public/open-aria#5]
Hello,

I'm currently working with the project and have encountered a dependency issue. The project depends on a closed-source library  https://github.com/mitre-public/open-aria/blob/ece0b3a221c24e1e85fd65ec27ddbe1f49f722c5/build.gradle.kts#L39 which I'm unable to find in any public Maven repositories.

Could you please provide guidance on how to obtain this JAR file? Alternatively, is there any workaround or solution that would allow me to compile the project without direct access to this dependency?

Thank you for your assistance.

**Comment by politeRaccoon** [comment]
@dragonlayout 
  Thank you for looking at OpenARIA.  This exact issue has top priority for this project.  The fix should be in by the end of the week if not earlier.  The fix will introduce a new "programming language agnostic" data format that permits the existing project to get rid of the closed source dependency.  

  I cannot provide the missing dependency because that work is not cleared for public release.  Additionally, it is a better long term target to integrate data format apis and data parsers directly in this project.

  I've already been working on this issue, and I hope to have a solution committed to main within the week.  I'll ping you when its done.

**Comment by mcrand29** [comment]
Hi, is there any update on this? I just tried to build the .jar file and the dependency on the closed-source library is still in place.

**Comment by politeRaccoon** [comment]
Yes, there is.  In [this commit](https://github.com/mitre-public/open-aria/commit/bd1446b1588c1f90321226da1258aae6aaf02210) we:

1. Introduce a spec for a new data format 
2. Begin the work to transitioning from the "old closed-source data format" to this "new open format".  The core issue was a dependency was not in the open-source domain.  So we needed to propose and create an alternative.  This work is ongoing, but the hardest part is behind us.

**Comment by politeRaccoon** [comment]
The closed source dependency has been removed.  

Here is a list of upcoming priorities:

- Adding github based CI/CD!  (I'm quite happy to finally have the closed-source software dependency out of the build)
- Publishing pre-canned "demo datasets" 
- Substantial refactor of Point and Track to permit more natural access to raw data records 
- E.g., Refactoring legacy design from interfaces and inheritance to java records, data-centric architecture, and composition.
- Supporting additional data formats
- See also:  [ADR on supporting new formats](https://github.com/mitre-public/open-aria/blob/main/docs/ADRs/supportingNewFormats.md)
- See also:  [ADR on refactoring Point and Track](https://github.com/mitre-public/open-aria/blob/main/docs/ADRs/pointInterfaceCritique2.md)

This issue will be close when the CI/CD is hooked up, demo datasets are published, and the CLI-based demos are operational.

**Comment by politeRaccoon** [comment]
Update:  CI/CD is hooked up using GitHub Actions.

You should be able to build locally using instructions in README.  

Still on the todo list:  Publishing demo datasets and improving the CLI-based demos.

**Comment by politeRaccoon** [comment]
Initial demos are completed and operations.  A combined "NOP" or "CSV" demo is targeted for Issue#38
